tools/wt_verify/wt_verify.py

```diff
@@ -172,6 +237,129 @@ def parse_output():
+        plt.xlim(left=0,right=xlimit)
+        plt.xlabel("Offset")
+        plt.ylabel("Page type")
+        plt.legend()
+        plt.title(f"Distribution of each page type for {filename} - {checkpoint}")
+        plt.close()
+
+        img = mpld3.fig_to_html(fig)
+        imgs += img
+    return imgs
+
+def show_block_distribution(filename, data, bins=100):
+    """
+    data: dictionary generated by parse_dump_blocks() that contains information related to different
+    checkpoints.
+    """
+    imgs = ""
+    for checkpoint in data:
+        all_addr = []
+
+        # Concatenate the lists for each page type:
+        for page_type in data[checkpoint]:
+            all_addr += data[checkpoint][page_type]
+
+        # Create a list of offsets.
+        all_addr = list(map(itemgetter(0), all_addr))
+
+        fig, ax = plt.subplots(1, figsize=(15, 10))
+        ax.hist(all_addr, bins=bins)
+        plt.title(f"Distribution of blocks for {filename} - {checkpoint}")
+        plt.xlabel(f"Offset (bins={bins})")
+        plt.ylabel("Frequency")
+        plt.close()
+
+        img = mpld3.fig_to_html(fig)
+        imgs += img
+    return imgs
+
+
+def show_free_block_distribution(filename, data, max_gap_size=0):
+    """
+    data: dictionary generated by parse_dump_blocks() that contains information related to different
+    checkpoints.
+    max_gap_size: display free blocks up to the specified size in bytes
+    """
+    imgs = ""
+    for checkpoint in data:
+        all_addr = []
+
+        # Concatenate the lists for each page type:
+        for page_type in data[checkpoint]:
+            all_addr += data[checkpoint][page_type]
+
+        # Sort them by the addr_start which is the first element:
+        all_addr.sort(key=itemgetter(0))
+
+        # Now count all the gaps:
+        buckets = {}
+        for i, current_tupe in enumerate(all_addr):
+            if i > 0:
+                prev_tupe = all_addr[i - 1]
+                prev_tupe_end = prev_tupe[0] + prev_tupe[1]
+                gap = current_tupe[0] - prev_tupe_end
+                assert gap >= 0, f"Data is not sorted correctly"
+                if gap == 0:
+                    continue
+                # If the size of the free block is large enough, we may not have interest in it.
+                if max_gap_size and gap > max_gap_size:
+                    continue
+                if gap in buckets:
+                    buckets[gap] += 1
+                else:
+                    buckets[gap] = 1
+            else:
+                # Nothing to do if the first block is written at offset 0.
+                gap = current_tupe[0]
+                if gap > 0:
+                    buckets[gap] = 1
+
+        # Sort buckets by size.
+        buckets = dict(sorted(buckets.items()))
+
+        fig, ax = plt.subplots(1, figsize=(15, 10))
+        keys_str = [str(x) for x in buckets.keys()]
+        ax.bar(keys_str, list(buckets.values()))
+        ax.set_xticklabels(keys_str)
```

Comment:
Wasn't the plan to make these labels vertical for readability? I also can't see `max_gap_size` getting used anywhere.

Reply:
> Wasn't the plan to make these labels vertical for readability?

You are going to laugh. It was not working and I found a solution suggesting to alternate lines. But unfortunately it still does not work for me either. I have been using `max_gap_size` to limit the number of buckets.

`max_gap_size` is used [here](https://github.com/wiredtiger/wiredtiger/pull/10597/files#diff-52d63dfce4d3dc743fd23c0ab653d394f52d4c2401b4008b28025f3e4e06c682R333).

Reply:
Ha, I loved that "just swap the lines" trick. I see that it's read there, but nothing calls the function with the argument set.

Reply:
Yes I tried locally and ended up removing it as it did not work.
Added back in 529b9bb2dea2ecc9260b394b4c7e65fd1b5b464e.